src/reflect/scala/reflect/internal/Scopes.scala

```diff
@@ -418,6 +431,8 @@ trait Scopes extends api.Scopes { self: SymbolTable =>
      */
     def iterator: Iterator[Symbol] = toList.iterator
 
+    def reverseIterator: Iterator[Symbol] = new ScopeIterator(this)
```

Comment:
We should document that this when using `reverseIterator` is not safe to mutate the `Scope`. Some uses of `Scope#iterator` do mutate the scope during iteration.